### PR TITLE
[risk=low][RW-5720] Display CDR Version name string in Workspace navbar

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

Also added a cdrVersion() function to utils and updated Workspace About to remove a state variable

cdr version in navbar

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [x] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-nav-bar.tsx [hunk 1]
```diff
@@ -2,35 +2,58 @@ import {Component, Input} from '@angular/core';
...
 import * as fp from 'lodash/fp';
 import * as React from 'react';
 
-
 const styles = reactStyles({
   container: {
-    display: 'flex', alignItems: 'center', backgroundColor: colors.secondary,
-    fontWeight: 500, color: 'white', textTransform: 'uppercase',
-    height: 60, paddingRight: 16,
+    display: 'flex',
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-nav-bar.tsx:21]: no changes to any of these - just formatting

File: ui/src/app/pages/workspace/workspace-about.tsx [hunk 2]
```diff
@@ -244,7 +239,7 @@ export const WorkspaceAbout = fp.flow(withUserProfile(), withUrlParams(), withCd
           </h3>
           <div style={styles.infoBox} data-test-id='cdrVersion'>
             <div style={styles.infoBoxHeader}>Dataset</div>
-            <div style={{fontSize: '0.5rem'}}>{cdrVersion ? cdrVersion.name : 'Loading...'}</div>
+            <div style={{fontSize: '0.5rem'}}>{workspace ? cdrVersion(workspace, cdrVersionListResponse).name : 'Loading...'}</div>
```
Comment (ericsong) [inline ui/src/app/pages/workspace/workspace-about.tsx:242]: is checking for the workspace sufficient here? What if cdrVersionListResponse is still loading?

File: ui/src/app/pages/workspace/workspace-nav-bar.tsx [hunk 3]
```diff
@@ -81,6 +104,9 @@ export const WorkspaceNavBarReact = fp.flow(
     {activeTabIndex > 0 && navSeparator}
     {fp.map(tab => navTab(tab, restrictTab(props.workspace, tab)), tabs)}
     <div style={{flexGrow: 1}}/>
+    {workspace && <div data-test-id='cdr-version' style={{textTransform: 'none'}}>
```
Comment (ericsong) [inline ui/src/app/pages/workspace/workspace-nav-bar.tsx:107]: do we need this check for a workspace if we're already doing that in the parent component before rendering this?

File: ui/src/app/utils/index.tsx [hunk 4]
```diff
@@ -627,3 +630,7 @@ export function validateInputForMySQL(searchString: string): Array<string> {
   }
   return Array.from(inputErrors);
 }
+
+export function cdrVersion(workspace: Workspace, cdrs: CdrVersionListResponse): CdrVersion {
```
Comment (ericsong) [inline ui/src/app/utils/index.tsx:634]: can you add a verb to the function? I wasn't really sure what it did until I found the definition.

File: ui/src/app/pages/workspace/workspace-nav-bar.tsx (hunk 3 above)
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-nav-bar.tsx:107]: That was also my assumption, but workspace is initially undefined here.  I don't know why.

File: ui/src/app/pages/workspace/workspace-about.tsx (hunk 2 above)
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-about.tsx:242]: There's a comment on the `withCdrVersions()` HOC: "Rendering of the connected component is blocked on initial load of the CDR versions."

**Review by ericsong — APPROVED** [review]

File: ui/src/app/pages/workspace/workspace-about.tsx (hunk 2 above)
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-about.tsx]: Oops.  Looks like you were right: #4191

File: ui/src/app/pages/workspace/workspace-nav-bar.tsx (hunk 3 above)
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-nav-bar.tsx]: This line got dropped somehow.  Re-added in #4191

**Pull request merged**